### PR TITLE
Stripe

### PR DESCRIPTION
**Pull request by ajkatom**
### What this PR does (required):
added methods to create customer, charge, create product, create plan, and subscribe to plan
* might add unsubscribe and update payment once I have a clearer idea of how wit will be utilized

### CODE REVIEW

File: server/models/Sitter.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+const mongoose = require('mongoose');
+const profileSechma = require('./Profile');
+
+const Profile = mongoose.model(Profile, profileSechma);
+const options = { discriminatorKey: 'kind' };
+
+const Sitter = Profile.discriminator(
+  'Sitter',
+  new mongoose.Schema({
```
Comment (moffatethan) [inline server/models/Sitter.js:9]: We should base the branch of this PR off of `sitterSchema` so this file isn't showing but that PR is approved so just merge the sitter schema PR in and this file will go away 🙂

File: server/utils/stripe.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
+const stripe = require('stripe')(process.env.STRIPE_SECERT);
+const Booking = require('../models/Booking');
+const Sitter = require('../models/Sitter');
+
```
Comment (moffatethan) [inline server/utils/stripe.js:4]: Good job making utils for this! Will definitely be beneficial as we roll out Stripe

File: server/utils/stripe.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+};
+
+exports.chargeCustomer = async (id) => {
+  const sitter = await Sitter.findById(booking.sitterId);
+  const startDate = new Date(booking.start);
+  const endDate = new Date(booking.end);
+  const diff = Math.abs(startDate - endDate);
+  const hours = diff / 36e5;
+  const amount = (hours * sitter.rate).toFixed(2);
+  const processingFee = 5;
```
Comment (moffatethan) [inline server/utils/stripe.js:23]: I'm okay on this for now - going to create a ticket for integrating this with the payment id.

File: server/utils/stripe.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  stripe.charges.create({
+    amount: amount + processingFee,
+    currency: CURRENCY,
+    customer: id,
+  });
+};
+
+exports.createProduct = async () => {
+  const PRODUCT_NAME = 'Monthly Subscription';
+  const PRODUCT_TYPE = 'service';
```
Comment (moffatethan) [inline server/utils/stripe.js:34]: So - I don't think we'll do monthly subscriptions in this app. Or would we create pet sitters products at all

File: server/utils/stripe.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  const product = await stripe.products.create({
+    name: PRODUCT_NAME,
+    type: PRODUCT_TYPE,
+  });
+
+  console.log(product);
+  return product.id;
+};
+exports.createPlan = async (productId) => {
+  const PLAN_NICKNAME = 'Monthly Subscription Plan';
```
Comment (moffatethan) [inline server/utils/stripe.js:45]: Ditto to my comment above, this isn't really necessary to have

File: server/utils/stripe.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+exports.subscribeCustomerToPlan = async (customerId, planId) => {
+  const subscription = await stripe.subscriptions.create({
+    customer: customerId,
+    items: [{ plan: planId }],
+  });
+
+  console.log(subscription);
+  return subscription;
+};
```
Comment (moffatethan) [inline server/utils/stripe.js:70]: Because we are removing the subscriptions can we also delete this 🙂

**Review by moffatethan — CHANGES_REQUESTED** [review]
Overall good work, like the utilities file for Stripe. Will help out a lot, don't need the subscription stuff but the others looks great! 

Just those comments then after this is good to be approved.

File: server/models/Sitter.js (hunk 1 above)
Comment (ajkatom) [reply server/models/Sitter.js:9]: I branched of that just because I needed to import the sitter model

File: server/utils/stripe.js (hunk 4 above)
Comment (ajkatom) [reply server/utils/stripe.js:34]: in the ticket there is a link to subscriptions that's why I created these methods
but I will take them off

**Review by moffatethan — APPROVED** [review]

**Pull request merged**